pallets/nfts/src/lib.rs

```diff
@@ -1371,6 +1529,55 @@ pub mod pallet {
 			Self::do_clear_all_transfer_approvals(maybe_check_origin, collection, item)
 		}
 
+		/// Cancel all the collection approvals.
+		///
+		/// Origin must be Signed.
+		///
+		/// Arguments:
+		/// - `collection`: The collection whose approvals will be cleared.
+		/// - `witness_approvals`: Information on the collection approvals cleared. This must be
+		///   correct.
+		///
+		/// Emits `AllApprovalsCancelled` on success.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(22)]
+		#[pallet::weight(T::WeightInfo::clear_all_collection_approvals(*witness_approvals))]
+		pub fn clear_all_collection_approvals(
```

Comment:
Where is the protection of `ApprovalsLimit`?

Reply:
See https://github.com/paritytech/polkadot-sdk/blob/377bc3f830a63e277dd94fed490670d1be6d840f/substrate/frame/assets/src/lib.rs#L827

This makes the witness data unnecessary and the function simply removes the max amount of approvals that it can. Returned by the dispatchable is the amount of approvals are removed to calculate the correct weight that needs to be paid by the origin.

Perhaps the function should therefore also be called `clear_collection_approvals` because we can not assure that all can be removed in a single transaction.

The alternative is that the caller requires to provide the witness data but then this has to be checked whether it is not more than the `ApprovalsLimit`

Reply:
I would prefer the alternative approach.

Reply:
I'm not sure how big the `ApprovalsLimit` will be but perhaps it only makes things more difficult. In other words, with your approach someone might have to call this function a dozen times, while if we allow to remove the max amount possible it might only have to be called say 3 times.

Curious to hear why you prefer the alternative approach?

Reply:
Resolved following the inital approach of Daan similar to pallet-assets. Agree with removing up to `ApprovalsLimit` and not an extra witness_approvals as parameter. I personally feel like that approach is more standardized and cleaner. (based on what Daan said above as well)